reliability-scanner/api/v1alpha1/pod/probes/probes.go

```diff
@@ -0,0 +1,111 @@
+package probes
+
+import (
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/vmware-tanzu/sonobuoy-plugins/reliability-scanner/internal"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var checkName string = "probes"
+
+// QuerierSpec defines the Specification for a Querier.
+type QuerierSpec struct {
+}
+
+// Querier defines the query and set of checks.
+type Querier struct {
+	client *kubernetes.Clientset
+	Spec   *QuerierSpec `yaml:"spec"`
+}
+
+// AddtoRunner configures a runner with the Querier for this check.
+func (querier *Querier) AddtoRunner(runner *internal.Runner) {
+	runner.Queriers = append(runner.Queriers, querier)
+	runner.Logger.WithFields(log.Fields{
+		"check_name": checkName,
+		"phase": "add",
+	}).Info("complete")
+}
+
+// NewQuerier returns a new configured Querier.
+func NewQuerier(spec *QuerierSpec) (Querier, error) {
+	out := Querier{
+		Spec: spec,
+	}
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return out, err
+	}
+	out.client, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		return out, err
+	}
+	return out, nil
+}
+
+// Start runs the Querier.
+func (q Querier) Start(cfg *internal.QuerierConfig) {
+	cfg.Logger.WithFields(log.Fields{
+		"check_name": checkName,
+		"phase": "add",
+	}).Info(internal.CheckStartMsg)
+
+	checkItem := internal.ReportItem{
+		Name:   checkName,
+		Status: "passed",
+	}
+
+	pods, err := q.client.CoreV1().Pods("").List(cfg.Context, metav1.ListOptions{})
```

Comment:
Does this work without a namespace provided ?

Reply:
yup. Will pull back everything. I want to include a configuration option per-check to only target a namespace, or list of namespce.